docs/bsp.md

```diff
@@ -706,18 +706,25 @@ Response:
 trait SourcesResult {
   def items: List[SourcesItem]
 }
+
 trait SourcesItem {
   def target: BuildTargetIdentifer
   /** The text documents or and directories that belong to this build target. */
   def sources: List[SourceItem]
 }
+
 trait SourceItem {
   /** Either a text document or a directory. A directory entry must end with a forward
     * slash "/" and a directory entry implies that every nested text document within the 
     * directory belongs to this source item.
     */
   def uri: Uri
   
+  /** A kind denotes properties of the source item, such as whether it is file or directory, and generated.
+   * If no kinds are specified, the item referenced by `uri` should be assumed to be a regular directory. 
+   */
+  def kinds: List[SourceItemKind]  
+  
   /** Specifies whether uri denotes a directory or regular file. 
     * If this is not set, and the uri does not point to an existing file, it should be treated as directory by default. */
   def isDirectory: Option[Boolean]
```

Comment:
Should we remove this approach now?

Reply:
The isDirectory? Yes, I forgot to remove it.